bonus.py

```diff
@@ -0,0 +1,30 @@
+from datetime import datetime
+
+
+def read_line(file_path, line_num):
+    # Opens input file and returns the wanted line
+    try:
+        with open(file_path, 'r') as input_file:
+            content = input_file.readlines()
+            return content[line_num - 1]
+    except FileNotFoundError:
+        write_error_to_file(f"Couldn't open the file: {file_path}.")
+    except OSError:
+        write_error_to_file(f"The path '{file_path}' is invalid.")
+    except EOFError:
+        write_error_to_file(f"End of file! Nothing left to read")
+        input_file.close()
+    except IndexError:
+        write_error_to_file(f"The file only has {len(content)} rows!")
+        input_file.close()
+    except Exception as unknown:
+        write_error_to_file(unknown.__str__())
+        input_file.close()
```

Comment:
Close resources using finally clause - close the file
https://www.geeksforgeeks.org/try-except-else-and-finally-in-python/